src/InnerList.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+namespace gapple\StructuredFields;
+
+class InnerList implements TupleInterface, \IteratorAggregate
+{
+    use TupleTrait;
+
+    public function __construct($value, ?object $parameters = null)
+    {
+        $this->value = $value;
+
+        if (is_null($parameters)) {
+            $this->parameters = new \stdClass();
+        } else {
+            $this->parameters = $parameters;
+        }
+    }
+
+    public function getIterator()
+    {
+        return new \ArrayIterator($this->value);
+    }
```

Comment:
I'm not sure if `InnerList` itself should be iterable.  
It adds a slight convenience over accessing `$innerList[0]` or `$innerList->value`, and parallels with `OuterList` being iterable, but I'm not sure is really expected?

Reply:
Iterating over an `OuterList` directly makes sense since it doesn't have parameters and the untyped form is an array, but for an `InnerList` I think it's better to require explicitly accessing the items.